arangod/Agency/Store.h

```diff
@@ -66,7 +66,7 @@ enum CheckMode { FIRST_FAIL, FULL };
 class Agent;
 
 /// @brief Key value tree
-class Store {
+class Store : public arangodb::Thread {
```

Comment:
Better add a private `StoreThread` class that is friend of the Store class and in its run method it just calls `store.run().`